platform/android/MapboxGLAndroidSDK/src/main/res/values/attrs.xml

```diff
@@ -3,117 +3,134 @@
-            <flag name="center_horizontal" value="0x01" />
-            <flag name="fill_horizontal" value="0x07" />
-            <flag name="center" value="0x11" />
-            <flag name="fill" value="0x77" />
-            <flag name="clip_vertical" value="0x80" />
-            <flag name="clip_horizontal" value="0x08" />
-            <flag name="start" value="0x00800003" />
-            <flag name="end" value="0x00800005" />
+            <flag name="top" value="0x30"/>
+            <flag name="bottom" value="0x50"/>
+            <flag name="left" value="0x03"/>
+            <flag name="right" value="0x05"/>
+            <flag name="center_vertical" value="0x10"/>
+            <flag name="fill_vertical" value="0x70"/>
+            <flag name="center_horizontal" value="0x01"/>
+            <flag name="fill_horizontal" value="0x07"/>
+            <flag name="center" value="0x11"/>
+            <flag name="fill" value="0x77"/>
+            <flag name="clip_vertical" value="0x80"/>
+            <flag name="clip_horizontal" value="0x08"/>
+            <flag name="start" value="0x00800003"/>
+            <flag name="end" value="0x00800005"/>
         </attr>
-        <attr name="mapbox_uiAttributionMarginLeft" format="dimension" />
-        <attr name="mapbox_uiAttributionMarginTop" format="dimension" />
-        <attr name="mapbox_uiAttributionMarginRight" format="dimension" />
-        <attr name="mapbox_uiAttributionMarginBottom" format="dimension" />
-        <attr name="mapbox_uiAttributionTintColor" format="color" />
+        <attr name="mapbox_uiAttributionMarginLeft" format="dimension"/>
+        <attr name="mapbox_uiAttributionMarginTop" format="dimension"/>
+        <attr name="mapbox_uiAttributionMarginRight" format="dimension"/>
+        <attr name="mapbox_uiAttributionMarginBottom" format="dimension"/>
+        <attr name="mapbox_uiAttributionTintColor" format="color"/>
 
         <!-- Deprecated to use TextureView-->
-        <attr name="mapbox_renderTextureMode" format="boolean" />
+        <attr name="mapbox_renderTextureMode" format="boolean"/>
 
     </declare-styleable>
+
+    <declare-styleable name="mapbox_BubbleLayout">
+        <attr name="mapbox_bl_arrowWidth" format="dimension|reference"/>
+        <attr name="mapbox_bl_cornersRadius" format="dimension|reference"/>
+        <attr name="mapbox_bl_arrowHeight" format="dimension|reference"/>
+        <attr name="mapbox_bl_arrowPosition" format="dimension|reference"/>
+        <attr name="mapbox_bl_bubbleColor" format="color|reference"/>
+        <attr name="mapbox_bl_strokeWidth" format="dimension|reference"/>
+        <attr name="mapbox_bl_strokeColor" format="color|reference"/>
+
+        <attr name="mapbox_bl_arrowDirection" format="enum">
+            <enum name="left" value="0"/>
+            <enum name="right" value="1"/>
+            <enum name="top" value="2"/>
+            <enum name="bottom" value="3"/>
+        </attr>
```

Comment:
If you want to expose these attributes to the end user, we also need to update the `res-public/public.xml`. More information about this in https://developer.android.com/studio/projects/android-library.html#PrivateResources.